README.md

```diff
@@ -1,510 +1,135 @@
 
-## `csv` preparation
-The `csv` specifies the input images and the reference vector data that will be use during the training.
-Each row in the `csv` file must contain 5 comma-separated items:
-- input image file (tif)
-- metadata info (*more details to come*). Optional, leave empty if not desired. (see example below)
-- reference vector data (GeoPackage)
-- attribute of the GeoPackage to use as classes values
-- dataset (one of 'trn' for training, 'val' for validation or 'tst' for test) where the sample will be used. Test set is optional.  
+- Samples labeled "*trn*" as per above are used to train the neural network.
+- Samples labeled "*val*" are used to estimate the training error (i.e. loss) on a set of sub-images not used for training, after every epoch.
+- At the end of all epochs, the model with the lowest error on validation data is loaded and samples labeled "*tst*", if they exist, are used to estimate the accuracy of the model on sub-images unseen during training or validation.
 
-Each image is a new line in the csv file.  For example:  
+### Inference ([inference.py](inference.py))
+The inference phase allows the use of a trained model to predict on new input data.
+The final step in the process is to assign every pixel in the original image a value corresponding to the most probable class.
 
-```
-\path\to\input\image1.tif,,\path\to\reference\vector1.gpkg,attribute,trn
-\path\to\input\image2.tif,,\path\to\reference\vector2.gpkg,attribute,val
-\path\to\input\image3.tif,,\path\to\reference\vector2.gpkg,attribute,tst
-```
+> The training and inference phases currently allow the use of a variety of neural networks to perform classification and semantic segmentation tasks (see the list in [models](models/)).
 
-## images_to_samples.py
+## **Requirement**
+This project comprises a set of commands to be run at a shell command prompt.  Examples used here are for a bash shell in an Ubuntu GNU/Linux environment.
 
-The first phase of the process is to determine sub-images (samples) to be used for training, validation and, optionally, test.  Images to be used must be of the geotiff type.  Sample locations in each image must be stored in a GeoPackage.
+- [Python 3.6](https://www.python.org/downloads/release/python-360/), see the full list of dependencies in [requirements.txt](requirements.txt)
+- [mlflow](https://mlflow.org/)
+- [minicanda](https://docs.conda.io/en/latest/miniconda.html) (highly recommended)
+- nvidia GPU (highly recommended)
 
-To launch the program:  
+> The system can be used on your workstation or cluster and on [AWS](https://aws.amazon.com/).
 
+## **Installation**
+Those step are for your a workstation on Ubuntu 18.04 using miniconda.
+Set and activate your python environment with the following commands:  
+```shell
+conda create -n gpu_ENV python=3.6 -c pytorch pytorch torchvision
+conda activate gpu_ENV
+conda install -c conda-forge ruamel_yaml h5py fiona rasterio geopandas scikit-image scikit-learn tqdm
```

Comment:
This line got stuck in solving environment for me (similar to https://github.com/conda/conda/issues/7690). I had to
`conda install --override-channels -c main -c conda-forge ruamel_yaml h5py fiona rasterio geopandas scikit-image scikit-learn tqdm`
to make it work. Perhaps this could be useful to other people in the future?